src/Microsoft.AspNet.Mvc.Localization/HtmlLocalizer.cs

```diff
@@ -0,0 +1,415 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using System.Text;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Localization;
+using Microsoft.Framework.WebEncoders;
+
+namespace Microsoft.AspNet.Mvc.Localization
+{
+    /// <summary>
+    /// An <see cref="IHtmlLocalizer"/> that uses the <see cref="IStringLocalizer"/> to provide localized HTML content.
+    /// </summary>
+    public class HtmlLocalizer : IHtmlLocalizer
+    {
+        private IStringLocalizer _localizer;
+        private readonly IStringLocalizerFactory _localizerFactory;
+        private readonly IHtmlEncoder _encoder;
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="factory">The <see cref="IStringLocalizerFactory"/>.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizerFactory factory, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizerFactory = factory;
+            _encoder = encoder;
+        }
+
+        /// <summary>
+        /// Creates a new <see cref="HtmlLocalizer"/>.
+        /// </summary>
+        /// <param name="localizer">The <see cref="IStringLocalizer"/> to read strings from.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/>.</param>
+        public HtmlLocalizer([NotNull] IStringLocalizer localizer, [NotNull] IHtmlEncoder encoder)
+        {
+            _localizer = localizer;
+            _encoder = encoder;
+        }
+
+        /// <inheritdoc />
+        public virtual LocalizedString this[[NotNull] string key] => _localizer[key];
```

Comment:
It is inheriting so we can use the goodness of `IStringLocalizer`. If we are using `IHtmlLocalizer` outside of views, we would use `LocalizedString`